extlinks/organisations/tests.py

```diff
@@ -282,3 +282,20 @@ def test_latest_links_csv(self):
         self.assertContains(response, self.linkevent4.link)
 
         self.assertContains(response, self.linkevent1.username.username)
+
+    def test_bot_edits_form(self):
```

Comment:
This test doesn't currently pass. The reason is that when we generate the example link events to test with (https://github.com/WikipediaLibrary/externallinks/blob/26e8bebc0f91faa6cf4849a7490ee3a0fa4d6d61/extlinks/organisations/tests.py#L69), none of them are currently flagged as bots. As such, any filtering we do will have no effect.

Instead, we should modify those linkevents so that one or two of them is a bot edit, then adjust the numbers in this (and the other) test to correspond to the expected number of added or removed links.

Reply:
Hi! Thank you for your comments: I'm really excited to be able to contribute to a genuine feature in an open source project. I'm going to run this by you first just in case my idea is ~~stupid~~ wrong.

I figure adding "user_is_bot=True" to one of the link events, and then change "total_added" to 0 and "total removed" to 1 will do right? Because the test should be removing one bot edit?

And then I do the same for the other test?

Reply:
>I figure adding "user_is_bot=True" to one of the link events

This much is correct, but there's a slight misunderstanding here on what 'links added/removed' means. The tool tracks additions of links to Wikimedia projects. For any given organisation or program it will then show the total number of links added or removed to Wikimedia projects over a given time period (and with particular filters like the one you're adding). So 'links added' here means 'how many link events in the database that match these filters were additions'.

In the setup:
https://github.com/WikipediaLibrary/externallinks/blob/master/extlinks/organisations/tests.py#L56

we define 4 link events to use for testing. They have a variety of data so that we can then test various filtering of that data. This test, for example:
https://github.com/WikipediaLibrary/externallinks/blob/master/extlinks/organisations/tests.py#L158

filters over a date range, and because of how we defined the link events in setup, we expect to see a total of 2 link events that were additions and 0 that were removals.

For this task, you're right that you should add `user_is_bot=True` to one or two of the linkevents. Then you want to adjust the numbers that correspond to whether the remaining links (not bots) were additions or removals (e.g. `change=LinkEvent.ADDED`). Does that make sense?

Reply:
Yes it does! I had to think really hard about it but yeah. So any of the link events that AREN'T bots, you just count whether it is added or removed?